<launch>

	<arg name="type_in_goal" default="false"/>
    <!-- 生成随机地图，并发布全局、局部点云 -->
    <node pkg="prometheus_planning_sim" name="random_forest" type="random_forest" output="screen">
        <!-- 参数 -->
        <param name="init_pos_x" value="0.0" type="double"/>
        <param name="init_pos_y" value="0.0" type="double"/>
        <!-- 地图尺寸、障碍物数量、分辨率 -->
        <param name="map/x_size" value="20.0" type="double"/>
        <param name="map/y_size" value="20.0" type="double"/>
        <param name="map/z_size" value="3.0" type="double"/>
        <param name="map/resolution" value="0.1" type="double"/>
        <param name="map/pillar_num" value="80" type="double"/>
        <param name="map/circle_num" value="0" type="double"/>
        <param name="map/min_distance" value="2.0" type="double"/>
        <param name="pillar_shape/min_radius" value="0.4" type="double"/>
        <param name="pillar_shape/max_radius" value="0.7" type="double"/>
        <param name="pillar_shape/min_height" value="2.0" type="double"/>
        <param name="pillar_shape/max_height" value="3.0" type="double"/>
        <param name="circle_shape/min_radius" value="0.7" type="double"/>
        <param name="circle_shape/max_radius" value="0.5" type="double"/>
        <param name="circle_shape/min_height" value="0.7" type="double"/>
        <param name="circle_shape/max_height" value="0.8" type="double"/>
        <param name="circle_shape/theta" value="0.5" type="double"/>
        <param name="sensing/sense_range" value="5.0" type="double"/>
        <param name="sensing/sense_rate" value="2.0" type="double"/>
    </node>

    <!-- 无人机运动学模拟器 -->
    <node pkg="prometheus_planning_sim" name="uav_sim" type="uav_sim" output="screen">	
        <!-- 参数 -->
        <param name="init_pos_x" value="0.0" type="double"/>
        <param name="init_pos_y" value="0.0" type="double"/>
        <param name="init_pos_z" value="1.0" type="double"/>
        <param name="planner_type" value="0" type="int"/>
        <param name="color/a" value="1.0"/>    
        <param name="color/r" value="0.0"/>        
        <param name="color/g" value="0.0"/>        
        <param name="color/b" value="0.0"/>       
        <param name="robot_scale" value="1.0"/>
        <param name="mesh_resource" value="$(find prometheus_planning_sim)/meshes/hummingbird.mesh"/>
    </node>

    <!-- 启动全局规划算法 -->
    <arg name="topic_of_global_pcl" default="/prometheus/planning/global_pcl_sim"/>
    <node pkg="prometheus_global_planning" name="global_planner_main" type="global_planner_main" output="screen" launch-prefix="gnome-terminal --tab --">	
        <remap from="/prometheus/global_planning/global_pcl" to="$(arg topic_of_global_pcl)" />
        <!-- 参数 -->
        <param name="global_planner/algorithm_mode" value="1" type="int"/> <!-- 0: A*，1: Kinodyamic-A* -->
	<param name="global_planner/planner_enable" value="true" type="bool"/>
	<param name="global_planner/map_groundtruth" value="false" type="bool"/>
        <!-- 二维平面规划 -->
        <param name="global_planner/is_2D" value="false" type="bool"/>
        <!-- ２维高度，建议与起飞高度一致 -->
        <param name="global_planner/fly_height_2D" value="1.0" type="double"/>
        <!-- 停止距离 -->
        <param name="global_planner/safe_distance" value="0.01" type="double"/>
        <param name="global_planner/time_per_path" value="0.2" type="double"/>
        <param name="global_planner/replan_time" value="2.0" type="double"/>
        <param name="global_planner/map_input" value="0" type="int"/>
        <param name="global_planner/sim_mode" value="true" type="bool"/>
        <!-- 最大搜索步数 -->
        <param name="astar/allocate_num" value="10000" type="int"/>
        <!-- 启发式函数系数 -->
        <param name="astar/lambda_heu" value="2.0" type="double"/>
        <!-- 分辨率 -->
        <param name="map/resolution" value="0.1" type="double"/>
        <!-- 障碍物膨胀距离 -->
        <param name="map/inflate" value="0.4" type="double"/>
        <!-- 地图范围 -->
        <param name="map/origin_x" value="-20.0" type="double"/>
        <param name="map/origin_y" value="-20.0" type="double"/>
        <param name="map/origin_z" value="-0.5" type="double"/>
        <param name="map/map_size_x" value="40.0" type="double"/>
        <param name="map/map_size_y" value="40.0" type="double"/>
        <param name="map/map_size_z" value="3.0" type="double"/>
    </node>

	<!-- run the pub_goal.cpp -->
	<node if="$(arg type_in_goal)" pkg="prometheus_mission" type="pub_goal" name="pub_goal" output="screen" launch-prefix="gnome-terminal --tab --">	
	</node>

	<!-- run the ground_station_msg.cpp -->
	<node pkg="prometheus_station" type="ground_station_msg" name="ground_station_msg" output="screen" launch-prefix="gnome-terminal --tab --">	
	</node>

	<!-- 启动rviz,设为false可关闭 -->
	<arg name="visualization" default="true"/>
	<group if="$(arg visualization)">
		<node type="rviz" name="rviz" pkg="rviz" args="-d $(find prometheus_planning_sim)/launch/sim_rviz_config.rviz" />
	</group>
</launch>

